### DISCUSSION TITLE
How to skip Overhead?

### DISCUSSION THREAD

**Discussion by DevAM-Tools** [dotnet/BenchmarkDotNet#2590] [Q&A]
For `ShortRunJob`s the calculation of the overhead often takes longer than the calculation of the workload itself. In addition, the overhead of just a few nanoseconds is in many cases negligible anyway.

- Is it possible to skip the calculation of the overhead?

- Is it possible to cache and reuse the overhead of functions with the same signature?

In my case I have like 10 function with the same signature that will be run with a number of parameters. 
```CSharp
[Benchmark]
public void Foo()
{
}
```
For every iteration a new calculation of the overhead is performed that comes basically to the same result.

**Comment by adamsitnik** [comment]
> Is it possible to skip the calculation of the overhead?

Yes, you can read more about similar settings in our [docs](https://benchmarkdotnet.org/articles/configs/jobs.html?q=EvaluateOverhead#accuracy)

```cs
public static void Main(string[] args)
{
    var config = ManualConfig.CreateMinimumViable()
        .AddJob(Job.ShortRun
            .WithEvaluateOverhead(false)); //
        
    BenchmarkSwitcher.FromAssembly(typeof(Program).Assembly).Run(args, config);
}
```

> Is it possible to cache and reuse the overhead of functions with the same signature?

No, it's currently not supported and I really doubt that we ever do that.

If you are searching for ways to reduce the time it takes to run the benchmarks you can check this config that is used by the official .NET benchmarking repo:

https://github.com/dotnet/performance/blob/40478a4b4f9a8bc5f648456a8870ba8fb774498f/src/harness/BenchmarkDotNet.Extensions/RecommendedConfig.cs#L33-L37
